### PR TITLE
librbd: new 'write_zeroes' API methods to suppliment the `discard` APIs

### PR DESCRIPTION
**Pull request by dillaman**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by dillaman** [comment]
I will follow-up with a second PR that introduces a new `NO_DISCARD` flag to provide a way to thickly provision space.

File: src/librbd/librbd.cc [hunk 1]
```diff
@@ -2571,7 +2571,7 @@ namespace librbd {
 
     bool discard_zero = ictx->config.get_val<bool>("rbd_discard_on_zeroed_write_same");
     if (discard_zero && mem_is_zero(bl.c_str(), bl.length())) {
```
Comment (trociny) [inline src/librbd/librbd.cc:2573]: Not related, just wondering: why is `bl.is_zero()` not used here?

Comment (dillaman) [reply src/librbd/librbd.cc:2573]: Good catch -- looks like it's just left-over from the original write-same commit 3fe9ab5140f

Comment (dillaman) [reply src/librbd/librbd.cc]: Fixed.

**Pull request merged**